Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM ubuntu:xenial
+
+# System dependencies
+RUN apt-get update && apt-get install -y curl
+
+# Install node
+ENV NVM_DIR /usr/local/nvm
+ENV NODE_VERSION 6.11.3
+
+# Install nvm with node and npm
+RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.6/install.sh | bash \
+    && . $NVM_DIR/nvm.sh \
+    && nvm install $NODE_VERSION \
+    && nvm alias default $NODE_VERSION \
+    && nvm use default
+ENV NODE_PATH $NVM_DIR/v$NODE_VERSION/lib/node_modules
+ENV PATH      $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
+
+# Install node dependencies
+RUN npm install -g yarn
+
+# Set git commit ID
+ARG COMMIT_ID
+ENV COMMIT_ID=$COMMIT_ID
+RUN test -n "${COMMIT_ID}"
+
+# Import code, install code dependencies
+WORKDIR /srv
+ADD . .
+RUN yarn install
+RUN yarn build
```

Comment:
What do `yarn install` and `yarn build` do? Ultimately, we'd like our docker image to be as slim as possible. In our other sites, yarn is exclusively used to pull in build-time dependencies. So what we do is:

``` bash
./run build  # Build all the site's resources locally
docker build .  # Now copy only the bits we need into the image
```

This means that we don't need to do any installing in the image, and therefore the image doesn't need to run `yarn`. It probably still needs `node` because it will need to run an ExpressJS server.

This is why I think you should start by sorting out the `./run` script - so we work out how to build the site and which bits are needed before we package those up a docker image.